### PR TITLE
Move smid from aomp-extras to deviceRTL

### PR DESCRIPTION
**Pull request by JonChesterfield**
Leaves a change corresponding to https://github.com/ROCm-Developer-Tools/aomp/issues/73 for a subsequent patch

### CODE REVIEW

**Comment by JonChesterfield** [comment]
Tests passing locally, despite fixing the off by one bug.

File: openmp/libomptarget/deviceRTLs/amdgcn/src/amdgcn_smid.hip [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+   OFFSET       10:6    Range: 0..31
+   SIZE         15:11   Range: 1..32
+ */
+
+#define GETREG_IMMED(SZ, OFF, REG) (SZ << 11) | (OFF << 6) | REG
+
+/*
+  __smid returns the wave's assigned Compute Unit and Shader Engine.
+  The Compute Unit, CU_ID returned in bits 3:0, and Shader Engine, SE_ID in bits
+  5:4. Note: the results vary over time. SZ minus 1 since SIZE is 1-based.
```
Comment (ronlieb) [inline openmp/libomptarget/deviceRTLs/amdgcn/src/amdgcn_smid.hip]: since a 5 bit field is going to return 0..31, wouldn't we want to add 1 to get 1..32 ?

Comment (JonChesterfield) [reply openmp/libomptarget/deviceRTLs/amdgcn/src/amdgcn_smid.hip]: Yep, that sounds right. I don't see Aaron on the potential reviewers list, will send h an email. What do you think of committing the patch without the -1 in the mean time?

**Comment by JonChesterfield** [comment]
Dropped the -1 modification and associated comment, will follow up with Aaron re: the issue

File: openmp/libomptarget/deviceRTLs/amdgcn/src/amdgcn_smid.hip (hunk 1 above)
Comment (JonChesterfield) [reply openmp/libomptarget/deviceRTLs/amdgcn/src/amdgcn_smid.hip]: Post coffee, I think I agree with Aaron. The hardware is going to read five bits and interpret them as a number in [1, 32]. Assuming the mapping is linear, this must mean 00000 => 1, 00001 => 2, 00010 => 3, 00011 => 4, 00100 => 5. We have four CU's (assuming that comment is right), so want the encoding 00011 which is (SZ-1)

**Comment by JonChesterfield** [comment]
Folded the two commits as github's rebase doesn't do so, but that's made the comment thread slightly harder to find

**Comment by JonChesterfield** [comment]
This is safe to land as it's a NFC movement of code from aomp-extras to deviceRTL. Uses of smid() in general are suspect however, will continue to look into that.

**Pull request merged**